tee-worker/scripts/litentry/release/deploy.sh

```diff
@@ -1,736 +1,555 @@
+
+# ------------------------------
+# main()
+# ------------------------------
+
+function main {
+  # 0/ check if $USER has sudo
+  if sudo -l -U $USER 2>/dev/null | grep -q 'may run the following'; then
+    source "$SGX_SDK/environment"
+  else
+    echo "$USER doesn't have sudo permission"
+    exit 1
+  fi
+
+  # 1/ create folders if missing
+  sudo mkdir -p "$BASEDIR"
+  sudo chown $USER:$GROUPS "$BASEDIR" 
+  for d in "$LOG_BACKUP_BASEDIR" "$WORKER_BACKUP_BASEDIR" "$RELAYCHAIN_ALICE_BASEDIR" "$RELAYCHAIN_BOB_BASEDIR" \
+    "$PARACHAIN_ALICE_BASEDIR" "$WORKER_BASEDIR"; do
+    mkdir -p "$d"
+  done
+
+  # 2/ parse command lines
+  echo "Parsing command line ..."
+  while [ $# -gt 0 ]; do
+    case "$1" in
+      -h|--help)
+        display_help
+        exit 0
+        ;;
+      -b|--build)
+        BUILD=true
+        shift
+        ;;
+      -d|--discard)
+        DISCARD=true
+        shift
+        ;;
+      -c|--config)
+        WORKER_CONFIG="$(realpath -s $2)"
+        shift 2
+        ;;
+      -a|--only-worker)
+        ONLY_WORKER=true
+        shift
+        ;;
+      -x|--chain)
+        CHAIN="$2"
+        shift
+        ;;
+      -p|--parachain-port)
+        PARACHAIN_PORT="$2"
+        shift
+        ;;
+      -z|--parachain-host)
+        PARACHAIN_HOST="$2"
+        shift
+        ;;
+      -v|--copy-from-docker)
+        COPY_FROM_DOCKER=true
+        DOCKER_IMAGE="$2"
+        shift
+        ;;
+      --prod)
+        PRODUCTION=true
+        shift
+        ;;
+      generate|restart|upgrade-worker)
+        ACTION="$1"
+        shift
+        ;;
+      *)
+        echo "Error: unknown option or subcommand $1"
+        display_help
+        exit 1
+        ;;
+    esac
+  done
+
+  # 3/ sanity checks
+  if [ ! -f "$WORKER_CONFIG" ]; then
+    echo "Worker config not found: $WORKER_CONFIG"
+    exit 1
+  fi
+
+  WORKER_COUNT=$(cat "$WORKER_CONFIG" | jq '.workers | length')
+  echo "Worker count: $WORKER_COUNT"
+
+  # TODO: check flags conflict, e.g.
+  # - having `--discard` together with `upgrade-worker` doesn't make sense
+  # - `upgrade-worker` should ignore the `--only-worker` flag
+
+  # 4/ main business logic
+  case "$ACTION" in
+    generate)
+      backup_services
+      generate_services
+      exit
+      ;;
+    restart)
+      backup_logs
+      backup_workers
+      stop_services
+      prune
+      build
+      setup_working_dir
+      if [ "$ONLY_WORKER" = true ]; then
+        remove_clean_reset
+      fi
+      restart_services
+      exit
+      ;;
+    upgrade-worker)
+      # build the new worker, the code must be under $ROOTDIR/tee-worker already
+      build_worker
+      # update the schedule
+      set_scheduled_enclave
+
+      # wait until sidechain stalls
+      wait_for_sidechain
+      backup_workers
+      stop_worker_services
+      get_old_mrenclave
+      # TODO: actually we only need the copy-up
+      setup_working_dir
+      migrate_shard
+      remove_clean_reset
```

Comment:
So when we are upgrading worker we are only removing the `--clean-reset` flag, Shouldn't we also update the latest parentchain block to sync from, else it will always will be`0`

Reply:
It won't be from 0, it will continue from where it's left off. So a bunch of blocks behind the latest, but not a lot

Reply:
It has the metadata stored in light_client DB IIRC